src/som/interpreter/actors/AbstractPromiseResolutionNode.java

```diff
@@ -51,15 +63,19 @@ protected void chainPromise(final SResolver resolver,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
     SPromise promiseToBeResolved = resolver.getPromise();
-
+    boolean breakpointOnResolution = isBreakpointOnPromiseResolution;
     synchronized (promiseValue) {
       Resolution state = promiseValue.getResolutionStateUnsync();
       if (SPromise.isCompleted(state)) {
         resolvePromise(state, resolver, promiseValue.getValueUnsync(),
-           isBreakpointOnPromiseResolution);
+            breakpointOnResolution);
       } else {
         synchronized (promiseToBeResolved) { // TODO: is this really deadlock free?
-          promiseToBeResolved.setTriggerResolutionBreakpointOnUnresolvedChainedPromise(isBreakpointOnPromiseResolution);
+          // TODO: check this condition.
```

Comment:
if I don't add that condition there is a case where the flag isBreakpointOnResolution is not propagated, look at this conversation for more detail https://metaconc.slack.com/archives/C1ZK5N4FQ/p1489677673962877